File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        // find the insert position 

        int l =0;
        int n = nums.size();
        int h = n-1;

        if(target>nums[h]){
            return h+1;
        }

        if(target<nums[l]){
            return 0;
        }

        while(l<h){
            int mid = l + (h-l)/2;
             if(nums[mid]==target){
                return mid;
             }

             if(nums[mid]<target){
               l=mid+1;
             }
             else {
                h=mid;
             }
            
        }


        return l;
    }
};

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # search in the insert position 


        # tell the insert position of target in array 
        
        n= len(nums)
        low =0
        high = n-1


        if target > nums[n-1]:
            return n
        
        if target < nums[0]:
            return 0

        while low < high:
            mid = low + (high-low)//2

            if nums[mid]==target:
                return mid
            elif target<nums[mid]:
                high = mid
            else :
                low = mid + 1
        

        return low 


